Tests/Editor/Unit/Core/EditingContextManagerTests.cs

```diff
@@ -0,0 +1,171 @@
+using UnityEngine;
+using UnityEditor;
+using System.Collections.Generic;
+using NUnit.Framework;
+using UnityEditor.Experimental.EditorVR.Core;
+using UnityEditor.Experimental.EditorVR.Utilities;
+using UnityEditor.Experimental.EditorVR.Tools;
+
+namespace UnityEditor.Experimental.EditorVR.Tests.Core
+{
+    [TestFixture]
+    public class EditingContextManagerTests
+    {
+		GameObject go;
+		EditorVRContext context, context2;
+		EditingContextManager manager;
+		EditingContextManagerSettings settings, newSettings;
+		SetEditingContextImplementor contextSetter;
+
+		[OneTimeSetUp]
+		public void Setup()
+		{
+			manager = EditingContextManager.s_Instance;
+			go = new GameObject("context test object");
+			var transformTool = go.AddComponent<TransformTool>();
+			var createPrimitiveTool = go.AddComponent<CreatePrimitiveTool>();
+
+			context = ScriptableObject.CreateInstance<EditorVRContext>();
+			context.name = "Some Other Context";
+			context.m_DefaultToolStack = new List<MonoScript>();
+			context.m_DefaultToolStack.Add(MonoScript.FromMonoBehaviour(transformTool));
+			context.m_DefaultToolStack.Add(MonoScript.FromMonoBehaviour(createPrimitiveTool));
+
+			context2 = ScriptableObject.CreateInstance<EditorVRContext>();
+			context2.name = "Yet Another Context";
+			context2.m_DefaultToolStack = context.m_DefaultToolStack;
+
+			settings = ScriptableObject.CreateInstance<EditingContextManagerSettings>();
+			settings.defaultContextName = "Custom Default Context";
+			newSettings = ScriptableObject.CreateInstance<EditingContextManagerSettings>();
+			newSettings.defaultContextName = "New Custom Default Context";
+		}
+
+		[Test]
+		public void Initializes_WithDefaultContext()
+		{
+			Assert.AreEqual(manager.defaultContext, manager.currentContext);
+		}
+
+		[Test]
+		public void Initializes_ISetEditingContextMethods()
+		{
+			Assert.IsNotNull(ISetEditingContextMethods.getAvailableEditingContexts);
+			Assert.IsNotNull(ISetEditingContextMethods.getPreviousEditingContexts);
+			Assert.IsNotNull(ISetEditingContextMethods.setEditingContext);
+			Assert.IsNotNull(ISetEditingContextMethods.restorePreviousEditingContext);
+		}
+
+		[Test]
+		public void SetEditingContext_DoesNothing_IfNull()
+		{
+			var beginningContext = manager.currentContext;
+			manager.SetEditingContext(null);
+			Assert.AreEqual(beginningContext, manager.currentContext);
+		}
+
+		[Test]
+		public void SetEditingContext_SetsCurrentContext_IfNotNull()
+		{
+			manager.SetEditingContext(context);
+			Assert.AreEqual(context, manager.currentContext);
+		}
+
+		[Test]
+		public void RestorePreviousContext_SetsPreviousContextToCurrent()
+		{
+			var beginningContext = manager.currentContext;
+			manager.SetEditingContext(context2);
+			Assert.AreNotEqual(beginningContext, manager.currentContext);
+
+			manager.RestorePreviousContext();
+			Assert.AreEqual(beginningContext, manager.currentContext);
+		}
+
+		[Test]
+		public void LoadProjectSettings_IfAssetFound()

```

Comment:
Is this going to blow away our existing context manager settings if we run this on our local box?